core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts

```diff
@@ -284,8 +287,27 @@ export class TranslationModalComponent {
     for (const [i, originalElement] of filteredOriginalElements.entries()) {
       if (originalElement.nodeName !== filteredTranslatedElements[
         i].nodeName) {
+        this.elementViolation = (
+          '"' + filteredTranslatedElements[i].outerHTML + '"' + ' does ' +
+          'not match ' + '"' + originalElement.outerHTML + '"' + '.');
         return false;
       }
+      if (originalElement.children.length !== filteredTranslatedElements[
+        i].children.length) {
+        this.elementViolation = (
+          '"' + filteredTranslatedElements[i].outerHTML + '"' + ' does ' +
+          'not match ' + '"' + originalElement.outerHTML + '"' + '.');
+        return false;
+      }
+      for (let index = 0; index < originalElement.children.length; index++) {
```

Comment:
Can we use `forEach` here (can also get the index)?

Reply:
I used `Array.some()` is it fine?